students/faundez.Rmd

```diff
@@ -64,7 +64,195 @@ summary(d)
+
+The above differences mean that when writing chunks of R code, a path can work in the Console and fail when you go to “Knit” the Rmarkdown file (*.Rmd), or vice versa.
+
+So let’s review some basic commands for navigating directories in both shell commands and R commands.
+
+Bash Shell
+The bash shell is the most common Unix-based command shell, found in Linux and Mac machines. It gets emulated for Windows in the Git Bash Shell application when installing git. Natively, Windows uses the less powerful Windows DOS command prompt, which uses cd (for pwd and cd) and dir (instead of ls).
+
+# present working directory
+pwd
+
+# change working directory
+cd
+
+# list files
+ls
+
+# list files that end in '.jpg'
+ls *.jpg
+Note the use of the wildcard * to indicate any set of characters.
+
+R
+Now play with the same commands commented above, but in R.
+
+```{r}
+# present working directory
+getwd()
+
+# change working directory
+setwd('.')
+
+# list files
+list.files()
+
+# list files that end in '.jpg'
+list.files(pattern=glob2rx('*.jpg'))
+
+# file exists
+file.exists('test.png')
+```
+
+Look at the help for list.files() (?list.files or F1 with cursor over list.files() in editing window) to see that the pattern argument expects a regular expression and glob2rx() changes the wildcard or globbing pattern into a regular expression.
+
+To work on your students/<user>.Rmd, I recomend you get the Console and your Rmarkdown file using the same working directory:
+
+```
+setwd('students')
+```
+
+#Install Packages
+```{r}
+# list of packages
+pkgs = c(
+  'readr',        # read csv
+  'readxl',       # read xls
+  'dplyr',        # data frame manipulation
+  'tidyr',        # data tidying
+  'nycflights13', # test dataset of NYC flights for 2013
+  'gapminder')    # test dataset of life expectancy and popultion
+
+# install packages if not found
+for (p in pkgs){
+  if (!require(p, character.only=T)){
+    install.packages(p)
+  }
+}
+```
+
+The gapminder dataset is “an excerpt of the data available at Gapminder.org. For each of 142 countries, the package provides values for life expectancy, GDP per capita, and population, every five years, from 1952 to 2007” (CRAN). Gapminder was the brain child of Hans Rosling who famously gave the TED Talk: The best stats you’ve ever seen - Hans Rosling.
+
+#Readings
+These are the main R packages we’ll be learning about this week:
+
+readr: column types
+dplyr: introduction
+tidyr: tidy data
+dplyr & tidyr: data wrangling cheatsheet
+Reading CSV
+utils::read.csv
+Traditionally, you would read a CSV like so:
+```{r}
+d = read.csv('../data/r-ecology/species.csv')
+d
+head(d)
+summary(d)
+```
+
+#Multiple Variables
+Now let’s approach this code sequentially using base functions, ie natively loaded functions in R without need for additional libraries.
+
+```{r}
+# read in csv
+surveys = read.csv("C:/Users/Patricia/Desktop/Environmental Informatics/env-info/data/r-ecology/surveys.csv") 
```

Comment:
"relative" path that works on my computer like `"../data/r-ecology/surveys.csv"` preferred over this "absolute" path hard-wired to your machine